contrib/scripts/codepage_to_utf8.py

```diff
@@ -0,0 +1,36 @@
+#!/bin/python3
+"""
+Convert file from legacy codepage to Unicode UTF-8
+"""
+import argparse
+import os
+import sys
+
+parser = argparse.ArgumentParser(description="Convert file from legacy codepage to Unicode UTF-8",
+                                 formatter_class=argparse.RawTextHelpFormatter)
+parser.add_argument("filename", help="filename to convert")
+parser.add_argument("--cp", type=int, default=437, help="codepage to convert from (default: 437)")
+parser.add_argument("-o", "--output", type=str,
+                    help="output filename (default: overwrite input file)")
+args = parser.parse_args()
+
+path = os.path.dirname(__file__)
+
+if not args.output:
+    args.output = args.filename
+
+try:
+    with open(os.path.join
+                  (path, './' + args.filename), 'r', encoding='cp' + str(args.cp)) as input_file:
+        lines = input_file.read()
+        with open(os.path.join(path, './' + args.output), 'w', encoding='utf-8') as output_file:
+            output_file.write(lines)
+            output_file.close()
+except LookupError:
+    print(f"Error: Unknown encoding '{args.cp}'")
+    sys.exit(1)
+except UnicodeDecodeError:
```

Comment:
Is this the correct exception if an error happens when decoding from codepage 437?  It's obviously not Unicode but maybe Python throws this regardless?

Reply:
The LookupError exception will occur if python does not find the encoding. So for instance if you pass `--cp 1`, it will fail with that error.

The UnicodeDecodeError will occur if it cannot decode the input file as the codepage specified. This is typically because the file is a binary file.

Reply:
Yeah, the LookupError makes sense.  I was just wondering if it actually throws a Unicode error when attempting to decode from code page 437 (not Unicode).  Seems like a badly named exception if that's the case.  But if it works, then sure no problem.

Reply:
Indeed it does throw that exception.